datafusion/common/src/file_options/csv_writer.rs

```diff
@@ -0,0 +1,91 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! Options related to how csv files should be written
+
+use std::str::FromStr;
+
+use arrow::csv::WriterBuilder;
+
+use crate::{
+    config::ConfigOptions,
+    error::{DataFusionError, Result},
+    parsers::CompressionTypeVariant,
+};
+
+use super::StatementOptions;
+
+/// Options for writing CSV files
+#[derive(Clone, Debug)]
+pub struct CsvWriterOptions {
+    /// Struct from the arrow crate which contains all csv writing related settings
+    pub writer_options: WriterBuilder,
+    /// Compression to apply after ArrowWriter serializes RecordBatches.
+    /// This compression is applied by DataFusion not the ArrowWriter itself.
+    pub compression: CompressionTypeVariant,
+    /// Indicates weather WriterBuilder.has_header() is set to true.
+    /// This is duplicative as WriterBuilder also stores this information.
+    /// However, WriterBuilder does not allow public read access to the
+    /// has_header parameter.
+    pub has_header: bool,
+    // TODO: expose a way to read has_header in arrow create
```

Comment:
I filed https://github.com/apache/arrow-rs/issues/4735

```suggestion
    // TODO: expose a way to read has_header in arrow create
    // https://github.com/apache/arrow-rs/issues/4735
```